### PR TITLE
comment replies

### PR DESCRIPTION
**Pull request by ajacksified**
UI in place; API stubbed.

👓 @schwers

### CODE REVIEW

File: src/app/router/handlers/CommentsPage.js [hunk 1]
```diff
@@ -45,4 +49,30 @@ export default class CommentsPage extends BaseHandler {
...
+    if (!state.session.isValid) {
+      return dispatch(platformActions.setPage('/login'));
+    }
+
+    try {
+      dispatch(commentActions.reply(thingId, text));
+
+      // Go back to the state before the comment form was opened, if we can go
+      // back. Otherwise, redirect to the redirectUrl passed in.
+      if (history && state.platform.history.length > 2) {
```
Comment (ajacksified) [inline src/app/router/handlers/CommentsPage.js]: I think we should build back functionality into platform that works on both server and client - which also means implementing referrer in platform.

File: src/app/router/handlers/CommentsPage.js [hunk 2]
```diff
@@ -45,4 +49,30 @@ export default class CommentsPage extends BaseHandler {
...
+    if (!state.session.isValid) {
+      return dispatch(platformActions.setPage('/login'));
+    }
+
+    try {
+      dispatch(commentActions.reply(thingId, text));
+
+      // Go back to the state before the comment form was opened, if we can go
+      // back. Otherwise, redirect to the redirectUrl passed in.
+      if (history && state.platform.history.length) {
```
Comment (ajacksified) [inline src/app/router/handlers/CommentsPage.js]: We should build `back` functionality into platform that works on both server and client - which also means implementing referrer in platform.

File: src/app/router/handlers/CommentsPage.js [hunk 3]
```diff
@@ -45,4 +49,30 @@ export default class CommentsPage extends BaseHandler {
...
+        return history.go(-1);
+      } else {
+        // todo fix after implementing referrer
+        // dispatch(platformActions.navigateToUrl(METHODS.GET, '/'));
+      }
+    } catch (e) {
+      console.log('Error commenting');
```
Comment (ajacksified) [inline src/app/router/handlers/CommentsPage.js]: We should also add error handling to future sprint planning; toasts would be nice.

File: src/app/router/handlers/CommentsPage.js (hunk 2 above)
Comment (schwers) [inline src/app/router/handlers/CommentsPage.js]: Agreed, Subreddit subscribe needs it as well

File: src/app/router/handlers/CommentsPage.js [hunk 4]
```diff
@@ -45,4 +49,30 @@ export default class CommentsPage extends BaseHandler {
     dispatch(commentsPageActions.fetchCommentsPage(commentsPageParams));
     fetchUserBasedData(dispatch);
   }
+
+  async [METHODS.POST](dispatch, getState/*, utils*/) {
+    const { thingId, text } = this.bodyParams;
+
+    const state = getState();
+
+    if (!state.session.isValid) {
```
Comment (schwers) [inline src/app/router/handlers/CommentsPage.js]: This check isn't quite right, `await`'s inside functions called in `dispatchBeforeSession` don't actually 'wait' before the handler is run. So you can have a valid `accessToken` but be in the middle of refreshing. This should probably use `waitForState`

File: src/app/router/handlers/CommentsPage.js (hunk 3 above)
Comment (schwers) [inline src/app/router/handlers/CommentsPage.js]: toasts++

**Comment by schwers** [comment]
:fish:

**Pull request merged**